### ISSUE TITLE
Discrepancies in Help> Using Lua

### ISSUE THREAD

**Issue by Vijayanth-Reddy-K** [mpcjanssen/simpletask-android#692]
- In **Parameters** description of `onFilter` and `onGroup`, `fields.lists` is mislabelled `fields.lits`.

- `onTextSearch (taskText, caseSensitive) -> boolean` to be `onTextSearch (taskText, searchText, caseSensitive) -> boolean`?

- Defining `onGroup` in the Lua configuration wondow makes more sense as it is only useful in the main program window. The widgets do not show group labels. So defining it in the filter there does nothing, as far as I can see. But putting it in the configuration window implemented it properly.

**Comment by smichel17** [comment]
I'm pretty sure you are correct on the first two bullets.

If you define `onGroup` in the global config, then you're forfeiting the ability to have a different onGroup for each filter. I'm not sure how that interacts if you also define `onGroup` in a filter.

**Comment by mpcjanssen** [comment]
@Vijayanth-Reddy-K 

In the code, the grouping is specifically linked to the filter. The fact that the widgets don't show it will not mean it will never happen in the future. I will need to check what happens if you put an onGroup in the main Lua config now. Not sure.

I will fix the help.

**Comment by Vijayanth-Reddy-K** [comment]
@smichel17 @mpcjanssen:

I tried the following: After modifying my global configuration file as shown in [#153](https://github.com/mpcjanssen/simpletask-android/issues/153#issuecomment-264688081), I also added one of the two onGroup functions to sort by multiple tags after the definition of `multigroup_by_list_or_tag`.

After successfully executing the new configuration file (required an app restart due to the buggy behaviour), by default, the main window tasks were sorted by tags, with entries with multiple tags being grouped in a multiline string, etc. This was the default behaviour on opening the app. In addition, when I opened the script menu from the main window and inserted the script to group by multiline lists, the tasks were immediately filtered by list (saying `X (12/12) Filtered on: Script`) and clicking the X button immediately went back to grouping by multiple tags.

When I used the filter by multiple lists script in a widget, the same behaviour was observed. While there was no change in the widget per se, because the group names are not displayed in the widget,  using the title bar to go to the main window created a filter that grouped by list, and when cleared, reverted to grouping by multiple tags.

So, in summary, the onGroup function defined in the main program window becomes the default behaviour for grouping, and other onGroup functions from widgets or scripts from the main window overwrite this "default behaviour" until cleared.

Surprisingly, this is not the case with the onFilter function. Defining an onFilter function in the global configuration file to show only completed tasks does not seem to work (as described in the help manual),  but if you intend to let the current onGroup function behaviour remain..

```
You should define the onGroup function in the filter (not in the configuration). Defining it in the main configuration will not work, if the Filter script is empty, the onGroup function will be undefined.
```

That line needs to be changed in Help as well.
